src/app/[locale]/(main)/quizbank/[id]/components/quizbank-options.tsx

```diff
@@ -0,0 +1,130 @@
+"use client"
+
+import React, { useCallback, useMemo, useState } from "react"
+import CopyQuizBankDialog from "./copy-quizbank-dialog/copy-quizbank-dialog"
+import { toast } from "@/components/ui/use-toast"
+import { updateQuizBankAction } from "../../add/actions/add-quiz-bank-action"
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip"
+import { Button } from "@/components/ui/button"
+import { Icons } from "@/components/ui/icons"
+import EditQuizBank from "./edit-button"
+import { useTranslations } from "next-intl"
+
+type Props = {
+  quizBankVisibility: "Private" | "Public"
+  quizbankId: string
+  userCurrentClass: any
+  isOwnQuizBank?: boolean
+}
+
+const QuizBankActions = ({
+  isOwnQuizBank,
+  userCurrentClass,
+  quizbankId,
+  quizBankVisibility,
+}: Props) => {
+  const i18n = useTranslations("ViewQuizBank")
+  const i18N = useTranslations("Change_quizbank_visibility")
+  const iErrors = useTranslations("Errors")
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
+  const toggleQuizBankVisibility = useCallback(async () => {
+    const togglevalue = quizBankVisibility === "Private" ? "Public" : "Private"
+    const result = await updateQuizBankAction(
+      { visibility: togglevalue },
+      quizbankId.toString()
+    )
+    if (!result.ok) {
+      setIsLoading(false)
+      return toast({
+        title: i18N("message.failed.title"),
+        description: iErrors(result.message),
+        variant: "flat",
+        color: "danger",
+      })
+    } else {
+      setIsLoading(false)
+      const visibility = i18N(`visibility.${togglevalue}`)
+      return toast({
+        title: i18N("message.success.title"),
+        description: i18N("message.success.description", {visibility}),
+        variant: "flat",
+        color: "success",
+      })
+    }
+  }, [i18N, iErrors, quizBankVisibility, quizbankId])
+
+  const quizBankActions = useMemo(() => {
+    const defaultButtons = [
+      <CopyQuizBankDialog
+        quizbankId={quizbankId}
+        buttonContent={i18n("author.copy_button")}
+        classes={userCurrentClass}
+      />,
+    ]
+    const OwnerQuizBankButtons = [
+      <EditQuizBank
+        quizbankId={quizbankId}
+        content={i18n("author.edit_button")}
+      />,
+      <Tooltip>
+        <TooltipTrigger asChild>
+          <Button
+            color="accent"
+            isIconOnly
+            onClick={() => {
+              setIsLoading(true)
+              toggleQuizBankVisibility()
```

Comment:
For better UX, please make an alert dialog to confirm changes. i18n term you can make up. 
P/s: https://ui.shadcn.com/docs/components/alert-dialog